src/HdbEventSubscriber.cpp

```diff
@@ -500,6 +500,8 @@ void HdbEventSubscriber::always_executed_hook()
 
 		if (state==Tango::ON)
 			set_status("Everything is OK");
+		else if(state==Tango::FAULT)
+			set_status("Error opening AttributeList file: " + attributeListFile);
```

Comment:
Right now this is the only case of fault but we should discriminate maybe, we can imagine a fault which will not be from the attribute list.
We should have this at init level as well, by reviewing it I saw a bug there as well, will comment on it.

Reply:
Well I don't know how to do it… But line 296 of HdbEventSubscriber.cpp, we check on hdb_dev->status, which I don't see used anywhere except here…
This would be nice to check for an error in reading the file there.

Reply:
It looks like hdb_dev->status was used in the past to set errors from the backend, but no more.
Going to commit a fix removing unused hdb_dev->status, setting fault at init also and not using FAULT state to decide if it is a file error.

Reply:
ok, thank you very much.
One last thing is that as it is the fault from reading the file supersedes any alarm. I think it is ok as it is a fault, and, as such, is more important than an alarm, but we could keep the information from both. Both options are fine by me, so as you want, and it's good to merge for me !

Reply:
I think a file FAULT prevent from working (and so prevent ALARM as well which can happen only when attribute list is loaded). So no way to make it work until file opening issue is solved, I would leave like this.